backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java

```diff
@@ -95,62 +97,67 @@ void applyInitialHubIdAndLicense(String initialId, String initialLicense) {
-				var client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
-				refreshLicense(refreshUrl.get(), license.getToken(), client);
+	@Scheduled(cron = "0 0 1 * * ?", timeZone = "UTC", concurrentExecution = Scheduled.ConcurrentExecution.SKIP, skipExecutionIf = LicenseHolder.class)
+	void refreshLicense() throws InterruptedException {
+		randomMinuteSleeper.sleep(); // add random sleep between [0,59]min to reduce infrastructure load
+		var refreshUrlClaim = get().getClaim("refreshUrl");
+		if (refreshUrlClaim != null) {
+			try {
+				var refreshUrl = URI.create(refreshUrlClaim.asString());
+				var refreshedLicense = requestLicenseRefresh(refreshUrl, get().getToken());
+				set(refreshedLicense);
+			} catch (LicenseRefreshFailedException lrfe) {
+				LOG.errorv("Failed to refresh license token. Request to {0} was answerd with response code {1,number,integer}", refreshUrlClaim, lrfe.statusCode);
+			} catch (IllegalArgumentException | IOException e) {
+				LOG.error("Failed to refresh license token", e);
+			} catch (JWTVerificationException jve) {
+				LOG.error("Failed to refresh license token. Refreshed token is invalid.", jve);
 			}
 		}
 	}
 
 	//visible for testing
-	void refreshLicense(String refreshUrl, String license, HttpClient client) throws InterruptedException {
-		var parameters = Map.of("token", license);
-		var body = parameters.entrySet() //
-				.stream() //
-				.map(e -> e.getKey() + "=" + URLEncoder.encode(e.getValue(), StandardCharsets.UTF_8)) //
-				.collect(Collectors.joining("&"));
-		var request = HttpRequest.newBuilder() //
-				.uri(URI.create(refreshUrl)) //
-				.headers("Content-Type", "application/x-www-form-urlencoded") //
-				.POST(HttpRequest.BodyPublishers.ofString(body)) //
-				.version(HttpClient.Version.HTTP_1_1) //
-				.build();
-		try {
+	String requestLicenseRefresh(URI refreshUrl, String licenseToken) throws InterruptedException, IOException, LicenseRefreshFailedException {
+		try (var client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build()) {
+			var body = "token=" + URLEncoder.encode(licenseToken, StandardCharsets.UTF_8);
+			var request = HttpRequest.newBuilder() //
+					.uri(refreshUrl) //
+					.headers("Content-Type", "application/x-www-form-urlencoded") //
+					.POST(HttpRequest.BodyPublishers.ofString(body)) //
+					.version(HttpClient.Version.HTTP_1_1) //
+					.build();
 			var response = client.send(request, HttpResponse.BodyHandlers.ofString());
 			if (response.statusCode() == 200 && !response.body().isEmpty()) {
-				set(response.body());
+				return response.body();
 			} else {
-				LOG.error("Failed to refresh license token with response code: " + response.statusCode());
+				throw new LicenseRefreshFailedException(response.statusCode(), body);
 			}
-		} catch (IOException | JWTVerificationException e) {
-			LOG.error("Failed to refresh license token", e);
 		}
 	}
 
+	//necessary for skipExecutionIf
```

Comment:
would be more legible in a public static inner class with a proper name such as `LicenseRefreshSkipper`

Reply:
refactored in f135a31c19b88918de69e101cb6aa08df6ee8fbf